### PR TITLE
feat(so-chooser-component): Update service offering chooser component

### PR DESCRIPTION
**Pull request by ksendart**
#502

### CODE REVIEW

File: src/app/service-offering/service-offering-selector/service-offering-selector.component.ts [hunk 1]
```diff
@@ -37,46 +44,6 @@ export class ServiceOfferingSelectorComponent implements ControlValueAccessor {
...
-        .subscribe(customOffering => {
-          this.setCustomOffering(customOffering);
-          this.change.next(this.serviceOffering);
-        });
-    } else {
-      this.change.next(this.serviceOffering);
-    }
-  }
-
   public registerOnChange(fn): void {
```
Comment (qqcky) [inline src/app/service-offering/service-offering-selector/service-offering-selector.component.ts]: Do we need these methods of `ControlValueAccessor` if you got rid of implementation?

File: src/config/config.json [hunk 2]
```diff
@@ -169,5 +169,29 @@
     "pulse": false
   },
   "templateGroups": [],
-  "accountTagsEnabled": false
+  "accountTagsEnabled": true,
```
Comment (HeyRoach) [inline src/config/config.json]: This option should be equal false

File: src/app/reducers/service-offerings/redux/service-offerings.actions.ts [hunk 3]
```diff
@@ -28,6 +32,16 @@ export class LoadOfferingsResponse implements Action {
...
   }
 }
+export class UpdateCustomServiceOffering implements Action {
+  type = UPDATE_CUSTOM_SERVICE_OFFERING;
+  constructor(public payload:  ServiceOffering ) {
+  }
+}
+export class ServiceOfferingsFilterUpdate implements Action {
+  type = SERVICE_OFFERINGS_FILTER_UPDATE;
+  constructor(public payload: any ) {
```
Comment (HeyRoach) [inline src/app/reducers/service-offerings/redux/service-offerings.actions.ts]: Maybe you should get rid of 'any'

File: src/app/service-offering/service-offering-filter/service-offering-filter.component.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  templateUrl: 'service-offering-filter.component.html'
+})
+export class ServiceOfferingFilterComponent {
+  @Input() public classes: Array<ServiceOfferingClass>;
+  @Input() public selectedClasses: ServiceOfferingClass[];
+  @Input() public query: string;
+  @Input() public viewMode: string;
+  @Input() public groupings: Array<any>;
+  @Input() public selectedGroupings: Array<any>;
+  @Output() public groupingsChange = new EventEmitter();
```
Comment (zolotyx) [inline src/app/service-offering/service-offering-filter/service-offering-filter.component.ts]: please add types of emitters

File: src/app/service-offering/service-offering-filter/service-offering-filter.component.html [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      >
+        {{ getName(c) | translate }}
+      </mat-option>
+    </mat-select>
+  </mat-form-field>
+</div>
+
+<div class="filter-wrapper">
+  <cs-search
+    [(ngModel)]="query"
```
Comment (zolotyx) [inline src/app/service-offering/service-offering-filter/service-offering-filter.component.html]: use one-way binding

File: src/app/service-offering/service-offering-filter/service-offering-filter.component.html [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      [attr.cdkFocusRegionStart]="viewMode === ServiceOfferingType.custom">
+      {{ 'SERVICE_OFFERING.MODES.CUSTOM' | translate }}
+    </mat-button-toggle>
+  </mat-button-toggle-group>
+
+  <mat-form-field *ngIf="classes?.length">
+    <mat-select
+      multiple="true"
+      [placeholder]="'SERVICE_OFFERING.FILTERS.SELECT_CLASSES' | translate"
+      [(ngModel)]="selectedClasses"
```
Comment (zolotyx) [inline src/app/service-offering/service-offering-filter/service-offering-filter.component.html]: use one-way binding

File: src/app/reducers/service-offerings/redux/service-offerings.reducers.ts [hunk 7]
```diff
@@ -239,30 +292,65 @@ export const getAvailableOfferings = createSelector(
...
+          : getCustomOfferingWithParams(offering, paramTags);
       }).filter(item => filterByCompatibilityPolicy(item) && filterStorageType(item));
     } else {
       return [];
     }
   }
 );
 
-export const getAvailableOfferingsForVmCreation = createSelector(selectAll,
+export const classesFilter = (offering: ServiceOffering, tags: Tag[], classesMap: any) => {
```
Comment (zolotyx) [inline src/app/reducers/service-offerings/redux/service-offerings.reducers.ts]: remove any

File: src/app/reducers/service-offerings/redux/service-offerings.reducers.ts [hunk 8]
```diff
@@ -239,30 +292,65 @@ export const getAvailableOfferings = createSelector(
...
+  const tag = offering && tags.find(tag => tag.key === ServiceOfferingClassKey + '.' + offering.id);
```
Comment (zolotyx) [inline src/app/reducers/service-offerings/redux/service-offerings.reducers.ts]: fix indentation

File: src/app/reducers/service-offerings/redux/service-offerings.reducers.ts [hunk 9]
```diff
@@ -239,30 +292,65 @@ export const getAvailableOfferings = createSelector(
...
+    const classes = tag && tag.value.split(',');
+    const showGeneral = !!classesMap[DefaultServiceOfferingClassId];
+    return classes && classes.reduce((m, group) => (m || !!classesMap[group]), false)
```
Comment (zolotyx) [inline src/app/reducers/service-offerings/redux/service-offerings.reducers.ts]: use find instead of reduce

**Pull request merged**